src/main/java/org/opensearch/security/securityconf/DynamicConfigFactory.java

```diff
@@ -357,14 +359,35 @@ public String getHash(String user) {
         public List<String> getSecurityRoles(String user) {
             InternalUserV7 tmp = internalUserV7SecurityDynamicConfiguration.getCEntry(user);
 
+            if (tmp == null) {
+                return ImmutableList.of();
+            }
+
+            // Log opendistro_security_roles regardless of which path we take
+            if (tmp.getOpendistro_security_roles() != null && !tmp.getOpendistro_security_roles().isEmpty()) {
+                log.warn(
+                    "Deprecated configuration opendistro_security_roles set for: {} "
+                        + "opendistro_security_roles will not be used in favor of direct_security_roles.",
+                    user
+                );
+            }
+
             // Security roles should only contain roles that exist in the roles dynamic config.
             // We should filter out any roles that have hidden rolesmapping.
-            return tmp == null
-                ? ImmutableList.of()
-                : tmp.getOpendistro_security_roles()
+            if (tmp.getDirect_security_roles() != null && !tmp.getDirect_security_roles().isEmpty()) {
+                return tmp.getDirect_security_roles()
                     .stream()
                     .filter(role -> !isRolesMappingHidden(role) && rolesV7SecurityDynamicConfiguration.exists(role))
                     .collect(ImmutableList.toImmutableList());
+            }
+
+            if (tmp.getOpendistro_security_roles() != null && !tmp.getOpendistro_security_roles().isEmpty()) {
+                return tmp.getOpendistro_security_roles()
```

Comment:
nit: line 368 logger can be moved here.

Reply:
we need to log early so that we always log irrespective of ``direct_security_roles`` or ``opendistro_security_roles`` being used.